### PR TITLE
[URGENT] Fix NoSuchMethodException and fix broken twitter auth

### PR DESCRIPTION
**Pull request by SUPERCILEX**

### CODE REVIEW

**Comment by samtstern** [comment]
@SUPERCILEX I can't reproduce this NoSuchMethoException can I ask a few questions:

  * Are you seeing it in the demo app or your own app?
  * Are you compiling with or without proguard obfuscation?
  * What version of the support library are you using alongside FirebaseUI?

While I don't object to these changes, there seems to be something strange going on here.  @amandle get a crash when using support `25.x` and you get the same crash when using support `24.x` and each of you thinks its fixed by changing the support library to the other version.

File: auth/README.md [hunk 1]
```diff
@@ -57,6 +57,11 @@ repositories {
 }
 ```
 
+#### Important note
+
+Firebase-UI-auth v1.0.0 is incompatible with the v25 support libraries. Please upgrade to v1.0.1
+to use the latest version of the support libraries.
+
```
Comment (SUPERCILEX) [inline auth/README.md]: What do you think of this statement (assuming FirebaseUI v1.0.1 contains this PR).

**Comment by samtstern** [comment]
@SUPERCILEX I am seeing some people with similar experiences around the internet:
http://stackoverflow.com/questions/40319860/com-android-supportcustomtabs-no-virtual-method-launchurl

So it seems the main thing we need to do is update the Facebook dependency (as you did in this PR).  I'd like @amandle to weigh in though since he dealt with the previous support library issue.

**Comment by samtstern** [comment]
Also can you confirm that if you **dont** enable custom tabs for Facebook (by not changing `fbYOUR_APP_ID`) that things work as expected with any version of the support library?

**Comment by SUPERCILEX** [comment]
@samtstern I just checked and facebook uses the webview as expected without `fbYOUR_APP_ID` configured.

**Comment by samtstern** [comment]
@SUPERCILEX ok *phew*.  That de-escalates this from "`1.0.0` is totally broken" to "there is a known issue with Facebook and Chrome Custom Tabs".  I am going to open an Issue about this to help people experiencing the crash

**Comment by SUPERCILEX** [comment]
@samtstern should I keep the note in the README?

**Comment by samtstern** [comment]
@SUPERCILEX I don't think it's necessary.  We've had releases before with some bad bugs and we've stuck with documenting them in issues.  This error does not seem world-breaking enough to warrant special treatment, although I appreciate the consideration.

**Comment by SUPERCILEX** [comment]
@samtstern Done.

**Comment by SUPERCILEX** [comment]
@samtstern @amandle Any idea when this will be merged?

**Comment by SUPERCILEX** [comment]
I found the root cause of the support library issue:
```
GradleCompatible: Incompatible Gradle Versions
../../../build.gradle: All com.android.support libraries must use the exact same version specification (mixing versions can lead to runtime crashes). Found versions 25.0.1, 25.0.0. Examples include com.android.support:animated-vector-drawable:25.0.1 and com.android.support:cardview-v7:25.0.0
Note: This issue has an associated quickfix operation in Android Studio/IntelliJ
Priority: 8 / 10
Category: Correctness
Severity: Error
Explanation: Incompatible Gradle Versions.
There are some combinations of libraries, or tools and libraries, that are incompatible, or can lead to bugs. One such incompatibility is compiling with a version of the Android support libraries that is not the latest version (or in particular, a version lower than your targetSdkVersion.)

More info:

To suppress this error, use the issue id "GradleCompatible" as explained in the Suppressing Warnings and Errors section.
```

**Comment by samtstern** [comment]
@SUPERCILEX sorry for the delay, @amandle and I have been crazy busy lately.  I am merging this now into `master`.  Still not sure about release timelines for vNext (which is either `v1.0.1 or v1.1.0`, depends on what goes in and what SemVer says).

**Comment by SUPERCILEX** [comment]
Sounds good! I've been able get around it by overriding the Gradle dependencies and the manifest.

**Pull request merged**